Gruntfile.js

```diff
@@ -4,42 +4,59 @@ const path = require('path');
 const webpack = require('webpack');
 const child_process = require('child_process');
 const TerserPlugin = require('terser-webpack-plugin');
+const CircularDependencyPlugin = require('circular-dependency-plugin');
+const VER = require('./tools/generate_version_file'); // Make the src/version.ts file.
 
-// A module entry file `src/entry/xxxx.ts` will be mapped to a build output file `build/xxxx.js`.
+const BUILD_VERSION = VER.VERSION;
+const BUILD_DATE = VER.DATE;
+const BUILD_ID = VER.ID;
+
+const CHECK_FOR_CIRCULAR_DEPENDENCIES = false;
```

Comment:
How come this isn't `true`?

Reply:
There are still some unresolved circular dependencies, I believe

Reply:
When set to true, it prints out a huge list of warnings during the build process. We could control this with an environment variable, like the VEX_DEVTOOL setting. 

For now, I just left it to false. If the build ever breaks due to a possible circular dependency, turn it on and it will help us debug. This is what I used to fix some of the circular dependencies that were breaking my build earlier.

Reply:
Okay, cool! Maybe better to rename this `DEBUG_CIRCULAR_DEPENDENCIES`.

Reply:
Done! Renamed the option to DEBUG_CIRCULAR_DEPENDENCIES. :-)